### PR TITLE
Keep chunks filenames in production mode

### PR DESCRIPTION
**Pull request by tusbar**
This allows specifying a custom chunk name, e.g.:

```js
const MyComponent = dynamic(import(/* webpackChunkName: "my-component" */ './my-component'), {
  ssr: false
})
```

and, in production, the chunk will keep its name (e.g.: `.next/static/chunks/my-component.7a942128037b1c30eb04.js`)

This is useful for stuff like bundle analyzer and makes it easier to know what’s requested. When a chunk name is not specified, it will use the default (0, 1, 2, etc.)

### CODE REVIEW

**Comment by timneutkens** [comment]
My main question would be what happens to `splitChunks`, I remember doing just contenthash because of splitchunks receiving a super long `[name]`. But that could be fixed by providing the `name` option inside `splitChunks` I guess 🤔

**Comment by tusbar** [comment]
Yeah, definitely, if you add a custom `splitChunk`, the generated name will be super long.

We could add a comment about that somewhere in the doc.

**Comment by tusbar** [comment]
@timneutkens how do you suggest we do that? loop over the splitchunks that don’t specify a name and add a generated name?

File: build/webpack.js [hunk 1]
```diff
@@ -213,7 +213,7 @@ export default async function getBaseWebpackConfig (dir: string, {dev = false, i
       // Precompile react / react-dom for development, speeding up webpack
       dev && !isServer && new AutoDllPlugin({
         filename: '[name]_[hash].js',
-        path: './static/dll',
+        path: './static/development/dll',
```
Comment (timneutkens) [inline build/webpack.js]: I made this change to make it clear this is a development only thing.

**Comment by timneutkens** [comment]
I just checked out your code and added tests. I realized we don't have to do anything about my previous concern right now as we don't use the default splitChunk configuration.

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
Thank you very much for your contribution 🙏👍

**Pull request merged**